lib/sdr_client/deposit/request.rb

```diff
@@ -24,28 +24,50 @@ def initialize(label: nil,
       end
 
       def as_json
-        json = {
+        {
           type: type,
-          administrative: {
-            hasAdminPolicy: apo
-          },
-          identification: {
-            sourceId: source_id
-          },
-          structural: {
-            isMemberOf: collection,
-            hasMember: file_sets_as_json
-          }
-        }
-        json[:label] = label if label
-        json[:identification][:catkey] = catkey if catkey
-        json
+          administrative: administrative,
+          identification: identification,
+          structural: structural
+        }.tap do |json|
+          json[:label] = label if label
+        end
+      end
+
+      # @return [Request] a clone of this request with the uploads added
+      def with_uploads(uploads)
+        Request.new(label: label,
+                    apo: apo,
+                    collection: collection,
+                    source_id: source_id,
+                    catkey: catkey,
+                    type: type,
+                    uploads: uploads)
       end
 
       private
 
       attr_reader :label, :uploads, :source_id, :catkey, :apo, :collection, :type
 
+      def administrative
+        {
+          hasAdminPolicy: apo
+        }
+      end
+
+      def identification
+        json = { sourceId: source_id }
```

Comment:
In `#as_json` above you changed the structure from being like the one used now in `#identification` to one that uses `#tap`. Should this method use the same pattern? I don't have a clear favorite; either is fine by me. Or is there a reason to have them use different patterns I may be spacing on?

Reply:
I like to consider myself a champion of diversity